tools/topology/v2/attributes.conf

```diff
@@ -0,0 +1,55 @@
+# common attributes and attribute types. Attributes, defined here can then be
+# used by any blocks. E.g. the "format" attribute can be used by pipelines,
+# DAIs, etc.
+
+# we're using an "availability" parameter in attribute declaration instead of
+# adding ".optional" or ".mandatory" to the array name
+
+DefineType.format {
+	enum [
+		"s16le"
+		"s24le"
+		"s32le"
+		"float"
+	]
+}
+
+DefineType.time_domain {
+	enum [
+		"dma"
+		"timer"
+	]
+}
+
+DefineAttribute.tokens {
+	type array.tuple
+}
+
+DefineAttribute.format {
```

Comment:
but we will have several attributes of type "format." Without defining the type "format" we'll have to repeat the same enum every time. Even worse with much longer enums like widget types.

Reply:
This will be global and in a global pcm.conf header. PCM format will be used in lots of places so why would we have to repeat. Can you provide an example ?

Reply:
@lgirdwood Even in minimal examples we have two formats: a pipeline format and a DAI format, so we need at least two attributes of type "format" - see `pipeline_format` and `dai_format` in this PR